In [26]:
%load_ext autoreload
%autoreload 2
from machine_lib import * 
from mylib import *
# TSOps,
s = login()


The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload
b'{"user":{"id":"DH81864"},"token":{"expiry":14400.0},"permissions":["CONSULTANT","MULTI_SIMULATION","PROD_ALPHAS","REFERRAL","VISUALIZATION","WORKDAY"]}'


In [3]:
# 获取数据集准备喂给gpt
df = get_datafields(s, dataset_id = 'model31', region='JPN', universe='TOP1600', delay=1)


In [3]:
n=df.id+":"+df.description
print(','.join(n))

mdl31_acc_chg_1_day:The 1-day change of region-relative 1 to 100 percentile Accruals Component score. 100 is the best and 1 is the worst.,mdl31_acc_chg_1_fy:The 1-year change of region-relative 1 to 100 percentile Accruals Component score. 100 is the best and 1 is the worst.,mdl31_acc_chg_1_qtr:The 1 quarter change of region-relative 1 to 100 percentile Accruals Component score. 100 is the best and 1 is the worst.,mdl31_acc_prior_qtr:Prior quarter region-relative 1 to 100 percentile Accruals Component score. 100 is the best and 1 is the worst.,mdl31_acc_prior_y:Prior year region-relative 1 to 100 percentile Accruals Component score. 100 is the best and 1 is the worst.,mdl31_ass_tvr_last_qtr:Last quarter asset turnover,mdl31_ass_tvr_last_y:Last year asset turnover,mdl31_ass_tvr_prior_qtr:Prior quarter asset turnover,mdl31_ass_tvr_prior_y:Prior year turnover,mdl31_ass_tvr_t4q:Trailing 4 Quarters asset turnover,mdl31_bv_share_last_qtr:Last quarter book value per share,mdl31_bv_share_last_

ts_mean, ts_median, ts_max, ts_min, ts_sum, ts_std_dev, ts_returns, ts_scale, ts_max_diff, ts_min_diff, ts_rank, ts_corr, ts_covariance, ts_partial_corr, ts_product, ts_decay_exp_window, ts_decay_linear, ts_zscore, ts_entropy


In [27]:
ops='ts_mean, ts_median, ts_max, ts_min, ts_sum, ts_std_dev, ts_returns, ts_scale, ts_max_diff, ts_min_diff, ts_rank, ts_corr, ts_covariance, ts_partial_corr, ts_product, ts_decay_exp_window, ts_decay_linear, ts_zscore, ts_entropy'.split(',')
profit='mdl31_acc_chg_1_day, mdl31_acc_chg_1_fy, mdl31_acc_chg_1_qtr, mdl31_acc_prior_qtr, mdl31_acc_prior_y, mdl31_cf_chg_1_day, mdl31_cf_chg_1_fy, mdl31_cf_chg_1_qtr, mdl31_cf_prior_qtr, mdl31_cf_prior_y, mdl31_cf_wc_mm_last_y, mdl31_dps_last_qtr, mdl31_dps_last_y, mdl31_dps_t4q, mdl31_dty_mm_fcf, mdl31_erq_raw_prior_qtr, mdl31_erq_re_rk_chg_1_day, mdl31_erq_re_rk_chg_1_fy, mdl31_erq_re_rk_chg_1_qtr, mdl31_erq_re_rk_prior_qtr, mdl31_erq_re_rk_prior_y, mdl31_fcf_mm_last_y, mdl31_gm_pct_last_qtr, mdl31_gm_pct_last_y, mdl31_gm_pct_prior_qtr, mdl31_gm_pct_prior_y, mdl31_gm_pct_t4q, mdl31_oe_chg_1_day, mdl31_oe_chg_1_fy, mdl31_oe_chg_1_qtr, mdl31_oe_prior_qtr, mdl31_oe_prior_y, mdl31_om_pct_last_qtr, mdl31_om_pct_last_y, mdl31_om_pct_prior_qtr, mdl31_om_pct_prior_y, mdl31_om_pct_t4q, mdl31_q4t_mm_fcf, mdl31_roe_pct_last_qtr, mdl31_roe_pct_last_y, mdl31_roe_pct_prior_qtr, mdl31_roe_pct_prior_y, mdl31_roe_pct_t4q, star_eq_accruals_rank, star_eq_cashflow_rank, star_eq_ope_rank, star_eq_rank, star_eq_region_rank'.split(',')
asset='mdl31_ass_tvr_last_qtr, mdl31_ass_tvr_last_y, mdl31_ass_tvr_prior_qtr, mdl31_ass_tvr_prior_y, mdl31_ass_tvr_t4q, mdl31_bv_share_last_qtr, mdl31_bv_share_last_y, mdl31_bv_share_prior_qtr, mdl31_bv_share_prior_y, mdl31_dse_last_qtr, mdl31_dse_last_y, mdl31_dse_prior_qtr, mdl31_dse_prior_y, mdl31_gva, mdl31_pb_current, mdl31_pb_last_qtr, mdl31_pb_last_y, mdl31_pb_prior_qtr, mdl31_pb_prior_y, mdl31_pricesales_current, mdl31_pricesales_last_qtr, mdl31_pricesales_last_y, mdl31_pricesales_prior_qtr, mdl31_pricesales_prior_y'.split(',')
days='30,60,120,250,360'.split(',')

In [28]:
expStr='A(B/C, D)'
exps=[]
for A in ops:
    for B in profit:
        for C in asset:
            for D in days:
                exps.append(expStr.replace('A',A).replace('B',B).replace('C',C).replace('D',D)) 

                       

In [29]:
fo_alpha_list = []
init_decay=5
for alpha in exps:
    fo_alpha_list.append((alpha, init_decay))
pools = load_task_pool(fo_alpha_list, 10, 10)
print(pools[0])

[[('ts_mean(mdl31_acc_chg_1_day/mdl31_ass_tvr_last_qtr, 30)', 5), ('ts_mean(mdl31_acc_chg_1_day/mdl31_ass_tvr_last_qtr, 60)', 5), ('ts_mean(mdl31_acc_chg_1_day/mdl31_ass_tvr_last_qtr, 120)', 5), ('ts_mean(mdl31_acc_chg_1_day/mdl31_ass_tvr_last_qtr, 250)', 5), ('ts_mean(mdl31_acc_chg_1_day/mdl31_ass_tvr_last_qtr, 360)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_last_y, 30)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_last_y, 60)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_last_y, 120)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_last_y, 250)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_last_y, 360)', 5)], [('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_prior_qtr, 30)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_prior_qtr, 60)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_prior_qtr, 120)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_prior_qtr, 250)', 5), ('ts_mean(mdl31_acc_chg_1_day/ mdl31_ass_tvr_prior_qtr, 360)', 5), ('ts_mean(mdl31_acc_ch

In [ ]:
so_tracker = get_alphas("11-19", "11-24", 1.2, 0.8, "KOR", 100, "track")


In [48]:
multi_simulate(pools, 'MARKET', 'JPN', 'TOP1600', 963)


b'{"user":{"id":"DH81864"},"token":{"expiry":14400.0},"permissions":["CONSULTANT","MULTI_SIMULATION","PROD_ALPHAS","REFERRAL","VISUALIZATION","WORKDAY"]}'
 loc key error


NameError: name 'traceback' is not defined

In [ ]:
so_alpha_list=[]
region='KOR'
group_ops = ["group_neutralize", "group_rank", "group_normalize", "group_scale", "group_zscore"]

for t in so_tracker['next']:
        for alpha in get_group_second_order_factory([t[1]], group_ops, region):
            so_alpha_list.append((alpha,t[7]))


In [ ]:
print(len(so_alpha_list))

In [ ]:
so_pools = load_task_pool(so_alpha_list, 10, 10)


In [ ]:
multi_simulate(so_pools, 'MARKET', 'KOR', 'TOP600', 43)


In [32]:
# 1.58 sharpe, 1 fitness, "submit"参数
th_tracker = get_alphas("10-15", "11-27", 1.58, 1, "KOR", 5000, "submit")
print(len(th_tracker))

b'{"user":{"id":"DH81864"},"token":{"expiry":14400.0},"permissions":["CONSULTANT","MULTI_SIMULATION","PROD_ALPHAS","REFERRAL","VISUALIZATION","WORKDAY"]}'
0
['X8eqbYz', "group_rank( ts_rank( fnd94_is_oper_inc_q/ fnd94_bs_com_eq_par_q, 250),densify(bucket(rank(cap), range='0.1, 1, 0.1')))", 2.11, 0.0776, 2.0, 0.002898, '2024-11-25T08:19:55-05:00', 5]
['EeLAvvm', "group_rank( ts_rank( fnd94_is_net_inc_basic_q/ fnd94_rt_com_eq_assets_q, 250),densify(bucket(rank(fnd28_value_05480/close), range='0.2, 1, 0.2')))", 2.1, 0.0679, 1.95, 0.003162, '2024-11-25T07:56:47-05:00', 5]
['vv60red', "group_rank( ts_rank( fnd94_is_oper_inc_q/ fnd94_bs_com_eq_par_q, 250),densify(bucket(group_rank(cap,sector),range='0,1,0.1')))", 2.09, 0.0802, 1.98, 0.002801, '2024-11-25T08:19:56-05:00', 5]
['dRoqbeK', "group_rank( ts_rank( fnd94_is_net_inc_basic_q/ fnd94_rt_com_eq_assets_q, 360),densify(bucket(rank(fnd28_value_05480/close), range='0.2, 1, 0.2')))", 2.07, 0.0656, 1.94, 0.00333, '2024-11-25T08:19:52-05:00', 5

In [ ]:
print(len(th_tracker))

In [43]:
## 将get的alpha的id取出至stone_bag，用api check submission
stone_bag = []
for alpha in th_tracker['next'] + th_tracker['decay']:
    stone_bag.append(alpha[0])
print(len(stone_bag))
gold_bag = []
check_submission(stone_bag, gold_bag, 325)

2248
b'{"user":{"id":"DH81864"},"token":{"expiry":14400.0},"permissions":["CONSULTANT","MULTI_SIMULATION","PROD_ALPHAS","REFERRAL","VISUALIZATION","WORKDAY"]}'
325


KeyboardInterrupt: 

In [46]:
check_submission(stone_bag, gold_bag, 2245)

b'{"user":{"id":"DH81864"},"token":{"expiry":14400.0},"permissions":["CONSULTANT","MULTI_SIMULATION","PROD_ALPHAS","REFERRAL","VISUALIZATION","WORKDAY"]}'
2245
[]


[('A7mbm3Y', np.float64(0.5981)),
 ('jrxdo5O', np.float64(0.5867)),
 ('97mAnE9', np.float64(0.4892)),
 ('bdnN1Wl', np.float64(0.6144)),
 ('wEPwzO2', np.float64(0.655)),
 ('wEPLnAQ', np.float64(0.478)),
 ('wEPLnJd', np.float64(0.4493)),
 ('MLZkb8r', np.float64(0.4584)),
 ('Q9gEnjK', np.float64(0.4846)),
 ('PO8vwLM', np.float64(0.4505)),
 ('n8Q3Z5E', np.float64(0.4604)),
 ('vv6xP53', np.float64(0.6874)),
 ('ogrV3bl', np.float64(0.4527)),
 ('ogrV32E', np.float64(0.4598)),
 ('7nMxR02', np.float64(0.5896))]

In [47]:
gold_bag

[('A7mbm3Y', np.float64(0.5981)),
 ('jrxdo5O', np.float64(0.5867)),
 ('97mAnE9', np.float64(0.4892)),
 ('bdnN1Wl', np.float64(0.6144)),
 ('wEPwzO2', np.float64(0.655)),
 ('wEPLnAQ', np.float64(0.478)),
 ('wEPLnJd', np.float64(0.4493)),
 ('MLZkb8r', np.float64(0.4584)),
 ('Q9gEnjK', np.float64(0.4846)),
 ('PO8vwLM', np.float64(0.4505)),
 ('n8Q3Z5E', np.float64(0.4604)),
 ('vv6xP53', np.float64(0.6874)),
 ('ogrV3bl', np.float64(0.4527)),
 ('ogrV32E', np.float64(0.4598)),
 ('7nMxR02', np.float64(0.5896))]

[('', np.float64(0.5981)),
 ('', np.float64(0.5867)),
 ('', np.float64(0.4892)),
 ('', np.float64(0.6144)),
 ('', np.float64(0.655)),
 ('wEPLnAQ', np.float64(0.478)),--
 ('wEPLnJd', np.float64(0.4493)),x
 ('MLZkb8r', np.float64(0.4584)),x
 ('Q9gEnjK', np.float64(0.4846)),
 ('PO8vwLM', np.float64(0.4505)),
 ('n8Q3Z5E', np.float64(0.4604)),
 ('vv6xP53', np.float64(0.6874)),--
 ('ogrV3bl', np.float64(0.4527)),x
 ('ogrV32E', np.float64(0.4598)),
 ('7nMxR02', np.float64(0.5896))]